schematicbrushreborn-core/src/main/java/de/eldoria/schematicbrush/config/ConfigurationImpl.java

```diff
@@ -44,7 +50,22 @@ public void saveConfigs() {
 
     @Override
     public void reloadConfigs() {
-        presets = loadConfig(PRESET_FILE, null, false).getObject("presets", PresetRegistry.class, new PresetRegistryImpl());
+        Database database = new Database(Map.of("host",
+                Objects.requireNonNull(this.getConfig().getString("mysql.host"))
+                , "port", this.getConfig().getInt("mysql.port"), "database",
+                Objects.requireNonNull(this.getConfig().getString("mysql.database")), "user",
+                Objects.requireNonNull(this.getConfig().getString("mysql.user")), "password",
+                Objects.requireNonNull(this.getConfig().getString("mysql.password"))));
+        DataSource dataSource = null;
+
+        if(Objects.requireNonNull(this.getConfig().getString("mysql.type")).equalsIgnoreCase("mariadb")) {
+            dataSource = DataSourceProvider.initMariaDBDataSource(this.plugin, database);
+        } else if(Objects.requireNonNull(this.getConfig().getString("mysql.type")).equalsIgnoreCase("mariadb")) {
+            dataSource = DataSourceProvider.initMySQLDataSource(this.plugin, database);
+        }
+
+
+        presets = loadConfig(PRESET_FILE, null, false).getObject("presets", PresetRegistry.class, new DatabasePresetRegistry(dataSource, this.plugin));
```

Comment:
Seems like this disables the usage of flat files in general. What about ppl without a database?